examples/ethernet/ethernet/main/ethernet_main.c

```diff
@@ -32,92 +32,47 @@
 #include "tcpip_adapter.h"
 #include "nvs_flash.h"
 #include "driver/gpio.h"
-#include "tlk110_phy.h"
+
+extern void phy_update_config(eth_config_t *config);
 
 static const char *TAG = "eth_demo";
 
-#define DEFAULT_PHY_CONFIG (AUTO_MDIX_ENABLE|AUTO_NEGOTIATION_ENABLE|AN_1|AN_0|LED_CFG)
 #define PIN_PHY_POWER 17
 #define PIN_SMI_MDC   23
 #define PIN_SMI_MDIO  18
 
-void phy_tlk110_check_phy_init(void)
-{
-    while((esp_eth_smi_read(BASIC_MODE_STATUS_REG) & AUTO_NEGOTIATION_COMPLETE ) != AUTO_NEGOTIATION_COMPLETE)
-    {};
-    while((esp_eth_smi_read(PHY_STATUS_REG) & AUTO_NEGTIATION_STATUS ) != AUTO_NEGTIATION_STATUS)
-    {};
-    while((esp_eth_smi_read(CABLE_DIAGNOSTIC_CONTROL_REG) & DIAGNOSTIC_DONE ) != DIAGNOSTIC_DONE)
-    {};
-}
-
-eth_speed_mode_t phy_tlk110_get_speed_mode(void)
-{
-    if((esp_eth_smi_read(PHY_STATUS_REG) & SPEED_STATUS ) != SPEED_STATUS) {
-        return ETH_SPEED_MODE_100M;
-    } else {
-        return ETH_SPEED_MODE_10M;
-    }   
-}
-
-eth_duplex_mode_t phy_tlk110_get_duplex_mode(void)
-{
-    if((esp_eth_smi_read(PHY_STATUS_REG) & DUPLEX_STATUS ) == DUPLEX_STATUS) {
-        return ETH_MDOE_FULLDUPLEX;
-    } else {
-        return ETH_MODE_HALFDUPLEX;
-    }   
-}
-
-bool phy_tlk110_check_phy_link_status(void)
-{
-    return ((esp_eth_smi_read(BASIC_MODE_STATUS_REG) & LINK_STATUS) == LINK_STATUS );
-}
+// select phy device
+//extern eth_config_t lan8720_default_ethernet_phy_config;
+//#define PHY_CONFIG lan8720_default_ethernet_phy_config
+extern eth_config_t tlk110_default_ethernet_phy_config;
+#define PHY_CONFIG tlk110_default_ethernet_phy_config
 
-bool phy_tlk110_get_partner_pause_enable(void)
-{
-    if((esp_eth_smi_read(PHY_LINK_PARTNER_ABILITY_REG) & PARTNER_PAUSE) == PARTNER_PAUSE) {
-        return true;
-    } else {
-        return false;
-    }   
-}
+// uncomment for console debug messages
+#define DEMO_DEBUG 1
```

Comment:
This pattern is not really "the IDF way". Use ESP_LOGD or ESP_LOGI to set the debug levels based on how important you think the output is.

Users can override the logging level (either per-file or globally) in various ways as described here:
http://esp-idf.readthedocs.io/en/latest/api/system/log.html

As I said before, I don't feel strongly about whether this example output should be ESP_LOGD or ESP_LOGI. Whichever you think is more useful is fine.